src/wp-includes/class-wp-block.php

```diff
@@ -526,4 +610,49 @@ public function render( $options = array() ) {
 
 		return $block_content;
 	}
+
+	/**
+	 * Populates the block attributes.
+	 *
+	 * @since 6.6.0
+	 */
+	protected function populate_attributes() {
+		// Originally, attributes could only be populated if the $attributes dynamic property had not been initialized.
+		// Therefore, this method should exit if the $attributes property has already been initialized.
+		if ( array_key_exists( 'attributes', get_object_vars( $this ) ) ) {
+			return;
+		}
+
+		$this->attributes = isset( $this->parsed_block['attrs'] ) ?
+			$this->parsed_block['attrs'] :
+			array();
+
+		if ( ! is_null( $this->block_type ) ) {
+			$this->attributes = $this->block_type->prepare_attributes_for_render( $this->attributes );
+		}
+	}
+
+	/**
+	 * Checks whether a property is declared as public.
+	 *
+	 * @since 6.6.0
+	 *
+	 * @param string $class_property_name Name of the class property to check.
+	 * @return bool True if the property is public, false otherwise.
+	 */
+	protected static function check_if_public_class_property( $class_property_name ) {
+		// The Reflection API is not used here for performance reasons.
+		// As the list is hardcoded, all newly declared public properties should be added to this list manually.
+		$public_class_properties = array(
```

Comment:
What if a child class adds public class properties? How will this change impact those implementations?

Reply:
I supposed that the child classes would need to provide their own implementation of the `check_if_public_class_property()` method.
wpdirectory.net search [shows](https://www.wpdirectory.net/search/01J91DTPYHME5C0A2T9P06J166) there are no classes that extend `WP_Block`.
To be safe, I don't mind refactoring `check_if_public_class_property()` to use the Reflection API to get the public class properties.

Reply:
Update: I’ve been giving this a lot of thought.
After all, it might be simpler to use the Reflection API to retrieve an array of all public property names for this class and store it in a new protected class property, utilizing lazy initialization for performance optimization. This approach seems to be  the only one compatible with child classes while requiring minimal maintenance.